pkg/controller/dependences.go

```diff
@@ -293,6 +298,15 @@ func newDependencies(
 	} else {
 		klog.Info("no permission for storage classes, skip creating sc lister")
 	}
+	supported, err := utildiscovery.IsAPIGroupVersionResourceSupported(kubeClientset.Discovery(), "networking.k8s.io/v1", "ingresses")
+	if err != nil {
+		klog.Fatalf("check if networking.k8s.io/v1/ingresses is supported failed: %s", err)
```

Comment:
If the k8s client discovers failed, it will panic. Is it ok?

Reply:
change `NewDependencies` to return (_,  error) ?

Reply:
If do that, all ut must add code to handle error of `NewDependencies` 😢 , so I give up.

Reply:
seems no test code call `NewDependencies` , we can panic inside `NewFakeDependencies` if `newDependencies` return error ?

Reply:
> @KanShiori CI failed

Fix, need to add the resource to fake discovery client